flink-java/src/main/java/org/apache/flink/api/java/sampling/PoissonSampler.java

```diff
@@ -93,29 +98,59 @@ public boolean hasNext() {
 					}
 				}
 			}
-
-			private void moveToNextElement() {
-				while (input.hasNext()) {
+			
+			public int poisson_ge1(double p){
```

Comment:
Why implement a new method instead of reuse poissonDistribution.sample() as before?